pkg/boot/initrd_test.go

```diff
@@ -107,3 +108,69 @@ func TestCatInitrds(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateInitrd(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		files       []file
+		wantContent []string
+	}{
+		{
+			name: "two files",
+			files: []file{
+				{
+					name:    "file1",
+					content: []byte("foo"),
+				},
+				{
+					name:    "file2",
+					content: []byte("bar"),
+				},
+			},
+			wantContent: []string{"foo", "bar", "TRAILER!!!"},
+		},
+		{
+			name: "one file",
+			files: []file{
+				{
+					name:    "file1",
+					content: []byte("foo"),
+				},
+			},
+			wantContent: []string{"foo", "TRAILER!!!"},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			dir, err := os.MkdirTemp("", "testdir")
+			if err != nil {
+				t.Errorf("CreateInitrd(%v): MkdirTemp errored: %v", tt.files, err)
+			}
+			defer os.RemoveAll(dir)
+
+			files := make([]string, len(tt.files))
+			for i, f := range tt.files {
+				tmpFile, err := os.CreateTemp(dir, f.name)
+				if err != nil {
+					t.Errorf("CreateInitrd(%v): CreateTemp errored: %v", tt.files, err)
+				}
+				err = os.WriteFile(tmpFile.Name(), f.content, 0666)
+				if err != nil {
+					t.Errorf("CreateInitrd(%v): Writing to tmpfile errored: %v", tt.files, err)
+				}
+				files[i] = tmpFile.Name()
+			}
+			got, err := CreateInitrd(files...)
+			if err != nil {
+				t.Errorf("CreateInitrd(%v): Creating initrd failed: %v", tt.files, err)
+			}
+
+			s := fmt.Sprintf("%s", got)
+			for _, st := range tt.wantContent {
+				if !strings.Contains(s, st) {
```

Comment:
If `s` might contain extra content beyond what are in `tt.wantContent`, this check will still still pass.

Can you make this check as absolute check other than contains ? e.g.

 if s == {expected content string} { // expected content can be calculated from `tt.wantContent`
   
}

Reply:
I've thought about how to derive the actual contents of the initrd as well but the only way I know how to do an absolute check is to look at the current results right now and hardcode it into the test. I guess the current implementation is a more basic check for the correctness of the initrd based on the presence of features that I know should be present.

Reply:
Especially since I'm using temporary file/directories it'll be difficult to actually reconstruct the expected initrd contents since the names won't be deterministic.. I'm open to any suggestions

Reply:
> I've thought about how to derive the actual contents of the initrd as well but the only way I know how to do an absolute check is to look at the current results right now and hardcode it into the test. I guess the current implementation is a more basic check for the correctness of the initrd based on the presence of features that I know should be present.

If current implementation is good, i think that is ok, so that it still help prevent future regression.  I think this might be similar to cat initrd tests, if we look at it in the way that we create initrd for each file, and cat them together into one. 

> Especially since I'm using temporary file/directories it'll be difficult to actually reconstruct the expected initrd contents since the names won't be deterministic.. I'm open to any suggestions

we can probably ignore the name field in checks. From code, I thought we could find the name for each file by `tmpFile.Name()`, after we created the test file

``` 
tmpFile, err := os.CreateTemp(dir, f.name)
```

Reply:
It's a little more complicated than the cat initrd tests, because we're creating the actual initrd while the cat tests are mocking out the initrd contents with hardcoded values like "foo" and "bar" and testing to see if the other aspects of the cat operation works on there. 

> we can probably ignore the name field in checks. From code, I thought we could find the name for each file by tmpFile.Name(), after we created the test file

That's fair.

Though I do think that this test will ensure the created initrd is well formed.